.github/workflows/build-docs.yml

```diff
@@ -25,7 +25,7 @@ jobs:
         make gen-docs
         touch docs/_build/html/.nojekyll
     - name: Publish Docs
-      uses: JamesIves/github-pages-deploy-action@3.7.1
+      uses: JamesIves/github-pages-deploy-action@v4
```

Comment:
Pinning individual commit hashes is more important for 3rd party actions like this one than the official github actions/...